### PR TITLE
doc: changelog.md updates

### PR DESCRIPTION
**Pull request by Totktonada**
* Fixed a typo in a link anchor.
* Added an alternative process (to better highlight why we stick with our one).

### CODE REVIEW

**Review by igormunkin — APPROVED** [review]
@Totktonada, thanks for the changes! LGTM since they are trivial.

Side note: I believe we can fix errors and typos, when building the final version of the changelog at the moment git tag is set, but yes, our approach is more flexible and convenient (at least for me).

File: doc/changelogs.md [hunk 1]
```diff
@@ -151,3 +151,16 @@ will be most convenient for our users.
 [gfm]: https://guides.github.com/features/mastering-markdown/
 [keep_a_changelog]: https://keepachangelog.com/en/1.0.0/
 [gitlab_changelog]: https://docs.gitlab.com/ee/development/changelog.html
+
+## Alternatives
+
```
Comment (kyukhin) [inline doc/changelogs.md:156]: I see no reason to use such an approach. Especially after cooking few final logs. If we use entries multiple times and it contains typos (which is frequently the case), then we need to fix the final draft multiple times.

Please, provide some reasoning here.

Comment (Totktonada) [reply doc/changelogs.md:156]: I don't propose to replace our current process (I think it is the best for us, no reason to change). I just want to mention all possible alternatives.

Comment (kyukhin) [reply doc/changelogs.md:156]: I think this is not an alternative due to the reasons I've mentioned above.

Comment (Totktonada) [reply doc/changelogs.md:156]: How should I call the section to don't make it confusing? Alternatives considered?

Comment (kyukhin) [reply doc/changelogs.md:156]: I think we should not mention wrong ways.

Comment (Totktonada) [reply doc/changelogs.md:156]: Okay, thanks for the opinion. I still think that we should not only describe 'how' we doing things, but also 'why'. That's the common principle in development. The example from our [commit message suggestions](https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message):

> 8. Use the body to explain **what and why** vs. how.

**Comment by Totktonada** [comment]
* Renamed 'Alternatives' to 'Alternatives considered'.
* Added the explicit list of considered alternatives (reasons 'why not' are still in the 'Reasoning' section).
* Highlighted the problems we would have without ability to modify a changelog entry at any time (the scenario is suggested by @kyukhin).
* In the commit message:
  - Replaced 'a possible alternative' with 'a declined alternative'.
  - Added 'that' for grammar.
* Rebased on current master.

File: doc/changelogs.md (hunk 1 above)
Comment (Totktonada) [reply doc/changelogs.md:170]: I tried to make it as explicit as possible that this alternative is not a proposal to use it. See the list of changes below.

File: doc/changelogs.md [hunk 2]
```diff
@@ -151,3 +151,35 @@ will be most convenient for our users.
...
+More alternatives are below.
+
+### Store a changelog entry in a commit message
+
+This process is [used in librepo][librepo_changelog_process]. It should mostly
+work for us, but there are several downsides:
+
+* A developer should remember about the changelog when writting a commit
+  message.
```
Comment (kyukhin) [inline doc/changelogs.md:177]: Typo. writting -> writing.

File: doc/changelogs.md [hunk 3]
```diff
@@ -151,3 +151,35 @@ will be most convenient for our users.
...
+* There is no ability to modify a changelog entry if a mistake is found later.
+
+It is easy to forget to fix a mistake if we can do it only just before a
```
Comment (kyukhin) [inline doc/changelogs.md:180]: I'd change wording: "... if we can only do it ...".

File: doc/changelogs.md [hunk 4]
```diff
@@ -151,3 +151,35 @@ will be most convenient for our users.
...
+release. The ability to modify a changelog entry becomes even more important,
+when the same changelog entry spread across different release branches and
```
Comment (kyukhin) [inline doc/changelogs.md:182]: spread -> spreads. Also, consider comma before "and".

File: doc/changelogs.md [hunk 5]
```diff
@@ -151,3 +151,35 @@ will be most convenient for our users.
...
+  a release notes draft on GitHub.
+* Single file changelog.
+
+See the previous section for reasons why we didn't pick up something from this
+list.
+
+More alternatives are below.
+
+### Store a changelog entry in a commit message
+
```
Comment (kyukhin) [inline doc/changelogs.md:172]: Okay, looks like you are obsessed with mentioning this useless method. OK, let's add it with mentioned precautions.

File: doc/changelogs.md [hunk 6]
```diff
@@ -151,3 +151,35 @@ will be most convenient for our users.
...
+This process is [used in librepo][librepo_changelog_process]. It should mostly
+work for us, but there are several downsides:
+
+* A developer should remember about the changelog when writting a commit
+  message.
+* There is no ability to modify a changelog entry if a mistake is found later.
+
+It is easy to forget to fix a mistake if we can do it only just before a
+release. The ability to modify a changelog entry becomes even more important,
```
Comment (kyukhin) [inline doc/changelogs.md:181]: The comma is useless.

**Review by kyukhin — CHANGES_REQUESTED** [review]

File: doc/changelogs.md (hunk 2 above)
Comment (Totktonada) [reply doc/changelogs.md:177]: Thanks! Fixed.

File: doc/changelogs.md (hunk 3 above)
Comment (Totktonada) [reply doc/changelogs.md]: I rewrote it entirely.

File: doc/changelogs.md (hunk 4 above)
Comment (Totktonada) [reply doc/changelogs.md]: I rewrote this sentence entirely.

File: doc/changelogs.md (hunk 6 above)
Comment (Totktonada) [reply doc/changelogs.md]: Seems so. Anyway, I don't satisfied by this wording and made another attempt to rewrite it.

File: doc/changelogs.md (hunk 5 above)
Comment (Totktonada) [reply doc/changelogs.md:172]: > Prior to stick with current process we considered several alternatives.

> See the previous section for reasons why we didn't pick up something from this
> list.

> It should mostly work for us, but there are several downsides:

Is it enough?

File: doc/changelogs.md (hunk 1 above)
Comment (Totktonada) [reply doc/changelogs.md:170]: I think that with given precautions it should look good. I'll resolve this thread.

File: doc/changelogs.md (hunk 5 above)
Comment (kyukhin) [reply doc/changelogs.md:172]: I'd state explicitly: "We'd avoid using this approach due to the following drawbacks:"

Comment (Totktonada) [reply doc/changelogs.md:172]: Applied:

```diff
-This process is [used in librepo][librepo_changelog_process]. It should mostly
-work for us, but there are several downsides:
+This process is [used in librepo][librepo_changelog_process]. We'd avoid using
+this approach due to the following drawbacks:
```

**Review by kyukhin — APPROVED** [review]

**Pull request merged**